src/sentry/api/endpoints/auth_index.py

```diff
@@ -257,6 +259,14 @@ def delete(self, request: Request, *args, **kwargs) -> Response:
 
         Deauthenticate all active sessions for this user.
         """
+        # For signals to work here, we must promote the request.user to a full user object
+        request.user = _promote_request_api_user(request)
         logout(request._request)
         request.user = AnonymousUser()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+def _promote_request_api_user(request):
```

Comment:
@corps @RyanSkonnord what do you guys think of this? This is basically a hack because I need a real user ORM object to pass into django login/logout built-ins. Should we leave the orm user object on the request while we're in the control silo?

Reply:
Yeah, my intention was to expose a separate request attribute for the ORM user, but with a name and documentation string that made it very clear that it would only be accessed for _write_ cases in the control silo, and should not be used in general.

So this is approaching the same idea, I think my only feedback is to ask: can we generalize this and give it a better name for the other places we will eventually need real User ORM objects in the control silo?